A new Data visualization project using D3

AakanxuShah opened the pull request:
# Data Visualization Using D3.js 

### Our goal is to learn the basics of data visualization using D3 through this project. 

![screen shot 2016-05-17 at 5 02 41 pm](https://cloud.githubusercontent.com/assets/8707780/15343407/4622fe20-1c51-11e6-9892-24e8a394b549.png)


What is D3.js? 

For those who are not familiar with D3, D3.js is a JavaScript library for manipulating documents based on data. D3 helps you bring data to life using HTML, SVG and CSS. 

D3.js helps you attach your data to DOM (Document Object Model) elements. Then you can use CSS3, HTML, and/or SVG showcase this data. Finally, you can make the data interactive through the use of D3.js data-driven transformations and transitions.


## Project Explanation: 

### Scenario : 
There’s a class of Online students interacting with different topics of the subject and gives quiz on those topics. 
There are 15 topics and on each topic, we have numbers of student who have taken quiz and have scored in three categories: Low, Mid and High

### For example (Given data ) : 
Topic:'1’, low:4, mid:13, high:18 
Topic:’2’,  low:11, mid:12, high:6
Topic:'3’,low:12, mid:24, high:6 and so on. 

Note that Topic 1 has 4 + 13 + 8 = 35 student has taken the quiz and Topic 2 has 11+12+6 = 29 student and topic 3 has 42 students and so on. 

**We want to make an interactive Bar and Pie charts. For example, a mouse hover on one of the bars will change the pie chart accordingly and vice-versa.  
With Interactive combination of bar chart and pie chart, where bar chart shows number of students who have interacted with particular topic(taken the quiz)  and pie chart showing classification of that students performance in categories of ‘low, mid, high’, we can visualize our data and get more data analysis from it.** 


## Hints and Resources : 

### Step by Step instructions: 
 	
D3 Introduction: https://d3js.org where you can learn about Selections , Dynamic properties and Transitions in D3.js.

•In the main javascript function, write a function to handle Histogram ( Bar chart ) - Histogram will show the total number of students who has taken the quiz ( interacted with that topic ) for 15 topics 
(Here's the Tutorial : https://bost.ocks.org/mike/bar/  where you can learn how to make a bar chart using the D3 JavaScript library. First tutorial teaches how to make a bare-bones version in HTML, then a more complete chart in Scalable Vector Graphics (SVG), and lastly animated transitions between views.) 

•	Create SVG for histogram ( Resource : http://codepen.io/SundeepB/pen/CxveH ) 
•	Create function for x-axis mapping and add x-axis to histogram SVG
•	Create function for y-axis mapping and Create bars for histogram to contain rectangles and topic labels.
•	create the rectangles and topic labels 
•	create function to update the bars. This will be used by pie-chart

•	write a function to handle pieChart. – pie chart will have three slices – Low , Mid and High to represent scores 
 (Tutorial - http://zeroviscosity.com/d3-js-step-by-step/step-1-a-basic-pie-chart where you can learn how to make a pie chart, then transitions between views and how to create a legend  )
•	create svg for pie chart.
•	create function to draw the arcs of the pie slices - pie slices will be Low , Mid and high 
•	create a function to compute the pie slice angles.
•	Draw the pie slices.
•	create function to update pie-chart. This will be used by histogram.

•	calculate total frequency by segment for all topic.
•	calculate total frequency by state for all segment.


## Result of Data Analysis and what can we infer from Visualization  :

- Initial pie chart shows aggregate classification of all student score on all topic combined into three categories ‘low, mid ,high’ 

- Initial bar chart showing number of students who have interacted on that particular topic 

- Any Selected category from pie chart will update bar chart, showing number of students who have interacted on various topics having score belonging to that particular category .
Screenshots below shows the mouse hover on 'Mid' slice and 'High' slice of the pie chart respectively and for that particular slice , topic-wise bar charts and number of students. 

![screen shot 2016-05-17 at 5 13 53 pm](https://cloud.githubusercontent.com/assets/8707780/15343597/ecb77c60-1c52-11e6-85ef-8affa8fc37ce.png)


![screen shot 2016-05-17 at 5 14 05 pm](https://cloud.githubusercontent.com/assets/8707780/15343605/f0e2f436-1c52-11e6-890d-d1300a943b4f.png)



- Any selected bar chart’s bar will update pie chart showing classification of all students score on that particular topic into three categories Low , Mid and High. 
Screenshot below shows the mouse hover on Topic 2 and for that particular topic , how many student are in categories - Low , Mid and High 

![screen shot 2016-05-17 at 5 13 26 pm](https://cloud.githubusercontent.com/assets/8707780/15343594/e666f16a-1c52-11e6-814f-6adac8345ddf.png)

AakanxuShah commented:
For issue #541 -  Create article for new Data visualization projects 
Please feel free to let me know of any improvements or changes that needs to be done.

Rafase282 commented:
@AakanxuShah for these kind of projects, we don't want to provide a working solution but hints and resources on how to complete it.

AakanxuShah commented:
@Rafase282 : Thank you ! I have updated the above comment. I have added more resources and hints and removed the solution.  Please let me know if there is anything else I can do to improve it.

Rafase282 commented:
@AakanxuShah it would be great if you make the article look like the comment you have.

AakanxuShah commented:
@Rafase282 Done 👍

Rafase282 commented:
Looks great :+1: 

I'll leave it open for a while longer to allow other members of the team to check and provide feedback. Thank you @AakanxuShah and keep them coming!

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+
+Note that Topic 1 has 4 + 13 + 8 = 35 student has taken the quiz and Topic 2 has 11+12+6 = 29 student and topic 3 has 42 students and so on. 
+
+**We want to make an interactive Bar and Pie charts. For example, a mouse hover on one of the bars will change the pie chart accordingly and vice-versa.  
```
~~We want to make an interactive Bar and Pie chart~~ We want to make interactive Bar and Pie charts.

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+•	calculate total frequency by state for all segment.
+
+
+## Result of Data Analysis and what can we infer from Visualization  :
```
~~Visualization  :~~ Visualization:

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+
+### Scenario : 
+There’s a class of Online students interacting with different topics of the subject and gives quiz on those topics. 
+There are 15 topics and on each topic, we have numbers of student who have taken quiz and have scored in three categories: Low, Mid and High
```
~~numbers of student~~ a number of students

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+•	Create SVG for histogram ( Resource : http://codepen.io/SundeepB/pen/CxveH ) 
+•	Create function for x-axis mapping and add x-axis to histogram SVG
+•	Create function for y-axis mapping and Create bars for histogram to contain rectangles and topic labels.
+•	create the rectangles and topic labels 
```
~~create~~ Create

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+•	Create function for x-axis mapping and add x-axis to histogram SVG
+•	Create function for y-axis mapping and Create bars for histogram to contain rectangles and topic labels.
+•	create the rectangles and topic labels 
+•	create function to update the bars. This will be used by pie-chart
```
~~create~~ Create

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+
+•	write a function to handle pieChart. – pie chart will have three slices – Low , Mid and High to represent scores 
+ (Tutorial - http://zeroviscosity.com/d3-js-step-by-step/step-1-a-basic-pie-chart where you can learn how to make a pie chart, then transitions between views and how to create a legend  )
+•	create svg for pie chart.
```
~~create~~ Create

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+•	write a function to handle pieChart. – pie chart will have three slices – Low , Mid and High to represent scores 
+ (Tutorial - http://zeroviscosity.com/d3-js-step-by-step/step-1-a-basic-pie-chart where you can learn how to make a pie chart, then transitions between views and how to create a legend  )
+•	create svg for pie chart.
+•	create function to draw the arcs of the pie slices - pie slices will be Low , Mid and high 
```
~~create~~ Create

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+ (Tutorial - http://zeroviscosity.com/d3-js-step-by-step/step-1-a-basic-pie-chart where you can learn how to make a pie chart, then transitions between views and how to create a legend  )
+•	create svg for pie chart.
+•	create function to draw the arcs of the pie slices - pie slices will be Low , Mid and high 
+•	create a function to compute the pie slice angles.
```
~~create~~ Create

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+•	create function to draw the arcs of the pie slices - pie slices will be Low , Mid and high 
+•	create a function to compute the pie slice angles.
+•	Draw the pie slices.
+•	create function to update pie-chart. This will be used by histogram.
```
~~create~~ Create

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+•	Draw the pie slices.
+•	create function to update pie-chart. This will be used by histogram.
+
+•	calculate total frequency by segment for all topic.
```
~~calculate~~ Calculate

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+•	create function to update pie-chart. This will be used by histogram.
+
+•	calculate total frequency by segment for all topic.
+•	calculate total frequency by state for all segment.
```
~~calculate~~ Calculate

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+•	create the rectangles and topic labels 
+•	create function to update the bars. This will be used by pie-chart
+
+•	write a function to handle pieChart. – pie chart will have three slices – Low , Mid and High to represent scores 
```
~~write~~ Write

theflametrooper commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,83 @@
+
+
+
+- Any selected bar chart’s bar will update pie chart showing classification of all students score on that particular topic into three categories Low , Mid and High. 
```
Remove space before "Any"

AakanxuShah replied:
Thank you for your input @theflametrooper ! 
I have made the changes you suggested but I did't remove the space before 'Any' because removing the space was changing the formatting ( instead of circular bulletin points, it was showing '-' if I remove the space )  and it looked unattractive.
Please feel free to let me know any more improvements !

Rafase282 replied:
```diff
-
-
-
+
```

I think that is what he means by removing the space.

theflametrooper commented:
### **LGTM :+1:**

Just implement this suggestion: https://github.com/FreeCodeCamp/wiki/pull/974/files#r63655657

Rafase282 commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,84 @@
+
+![screen shot 2016-05-17 at 5 13 53 pm](https://cloud.githubusercontent.com/assets/8707780/15343597/ecb77c60-1c52-11e6-85ef-8affa8fc37ce.png)
+
+
```
```diff
-
-
+
```

Rafase282 commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,84 @@
+•	Calculate total frequency by segment for all topic.
+•	Calculate total frequency by state for all segment.
+
+
```
```diff
-
-
+
```

Rafase282 commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,84 @@
+For those who are not familiar with D3, D3.js is a JavaScript library for manipulating documents based on data. D3 helps you bring data to life using HTML, SVG and CSS. 
+
+D3.js helps you attach your data to DOM (Document Object Model) elements. Then you can use CSS3, HTML, and/or SVG showcase this data. Finally, you can make the data interactive through the use of D3.js data-driven transformations and transitions.
+
```
```diff
-
-
+
```

Rafase282 commented on Data-Visualization-Project.md:
```diff
@@ -0,0 +1,84 @@
+
+![screen shot 2016-05-17 at 5 02 41 pm](https://cloud.githubusercontent.com/assets/8707780/15343407/4622fe20-1c51-11e6-9892-24e8a394b549.png)
+
+
```
```diff
-
-
+
```

Rafase282 commented:
@AakanxuShah I added more clear instructions on what you need to change, after that I'll squash and merge.
